hana/enable_cost_optimized.sls

```diff
@@ -47,9 +47,24 @@ install_hana_python_packages:
       - group: sapsys
       - enforce_toplevel: False
       - source: {{ grains['hana_inst_folder']~'/DATA_UNITS/HDB_CLIENT_LINUX_X86_64/client/PYDBAPI.TGZ' }}
```

Comment:
Hi @Simranpal ,
I have found one thing here that we should change. Using this `grains` we totally depend in the terraform deployment, and we cannot do that.
We need to find other way of passing this `hana_inst_folder` data. It might be in the pillar file (i don't know if this could be autodetected somehow).
Let's find the best solution for that

Reply:
@Simranpal @arbulu89 Does this path work? The hana_inst_folder contains the NFS share location, so it would need to be mounted. This path should be composed using the pillar variable `software_path:` that contains the already mounted path of the SAP binaries.
Another important point is that the path only works for x86_64 platform. It should be composed based on the architecture being used, to work on Power too.

Reply:
@diegoakechi Here we assume that the files are already in that path. We have to consider that they might be downloaded because we are deploying in some cloud provider.
Other feature would be just to get the installation files in this formula.

Actually, we have had hardcoded even in shaptools this path. We need to make it variable even there (right now we are only using x86)

Reply:
> @diegoakechi Here we assume that the files are already in that path. We have to consider that they might be downloaded because we are deploying in some cloud provider.
The client is provided as part of the HANA Installer as a whole. So it should use the same directory that was mounted in order to perform the HANA installation, that is the value included on the `software_path:` entry on the `hana.sls` pillar file. 

> Other feature would be just to get the installation files in this formula.
> 
> Actually, we have had hardcoded even in shaptools this path. We need to make it variable even there (right now we are only using x86)
Yes, this is a problem but should be easily solved with something like `os.uname()` or `platform.machine()`.

Reply:
Hi @arbulu89 @diegoakechi I will use the pillar data entry("software_path") for fetching the client py files.

For the platform specific HANA client path, we need to store the platform-type first in grains? and then maybe based on that update the salt logic..

Reply:
@Simranpal Salt already gets it on the os core grains. I think the grains['cpuarch'] could be used for that. The problem is, that if we need to compose it in multiple places, maybe would make more sense to move it to shaptools as a method, or as a function inside the salt modules. What do you think @arbulu89?

Reply:
@diegoakechi @Simranpal Actually we will need to do this updates in shaptools too. By now, forget about the CPU arch, we will manage this in a next PR.
We need to be careful to with this, maybe managing this in the formula is not a good idea (neither in shaptools) we will see